src/modules/common/templates/icon.tsx

```diff
@@ -0,0 +1,21 @@
+'use client'
+
+import { tw } from "@/lib/utils/tw";
+import { IconProps } from "@/types/templates";
+import React from "react";
+
+const Icon: React.FC<IconProps> = ({ className, icon }) => 
+{
+    return (
+        <span
+         className={tw(
+            "font-bold size-4",
+            "text-light-text dark:text-dark-text",
+            className
+         )}>
+            {icon}
+        </span>
+    );
+}
+
+export default Icon;
```

Comment:
You should maintain a similar structure to the rest of the project, making use of `'` when you are not in an HTML-based structure and keep an order in the imports based on their size.

```suggestion
'use client'

import { IconProps } from '@/types/templates'
import { tw } from '@/lib/utils/tw'
import React from 'react'

const Icon: React.FC<IconProps> = ({ className, icon }) => 
{
    return (
        <span
         className={tw(
            "font-bold size-4",
            "text-light-text dark:text-dark-text",
            className
         )}>
            {icon}
        </span>
    )
}

export default Icon
```